lua/impatient/cachepack.lua

```diff
@@ -0,0 +1,90 @@
+local ffi = require('ffi')
+
+local M = {}
+
+-- using double for packing/unpacking numbers has no conversion overhead
+local c_double = ffi.typeof("double[1]")
+local sizeof_c_double = ffi.sizeof("double")
+
+local OutputBuffer = {}
+
+function OutputBuffer.create() return {} end
+
+function OutputBuffer.write_number(buf, num)
```

Comment:
I would remove the OutputBuffer and InputBuffer tables and just use local functions with slightly more descriptive names.

Table accesses cause a hash lookup as opposed to an index lookup and given that these functions are called in loops for hundreds of iterations (263 on my machine), the cost may be noticeable. Though it is likely that LuaJIT is caching the table lookups for us so this point is probably moot, but I think it's worth doing anyway.

Reply:
Originally I had them as local functions, I was just trying to keep it organized and forgot about the overhead (it's been a _really_ long time since I touched LuaJIT's internals)

Feel free to modify it to your liking :)